File: core/src/main/java/com/averude/uksatse/scheduler/core/model/dto/EmployeeWorkStatDTO.java
package com.averude.uksatse.scheduler.core.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class EmployeeWorkStatDTO {
    private Long employeeId;
    private Long shiftId;
    private Long mainPositionId;
    private List<EmployeePositionStat> positionStats;
}

File: frontend-server/src/app/model/dto/employee-work-stat-dto.ts
export class EmployeeWorkStatDTO {
  employeeId: number;
  shiftId: number;
  mainPositionId: number;
  positionStats: EmployeePositionStat[];
}

export class SummationResult {
  summationColumnId: number;
  type: string;
  value: number;
}

export enum SummationMode {
  OVERALL = "overall",
  PER_POSITION = "per_position"
}

export class EmployeePositionStat {
  positionId: number;
  summations: SummationResult[];
}
